extension/src/decoding.rs

```diff
@@ -0,0 +1,55 @@
+use super::*;
+use pallet_contracts::chain_extension::BufIn;
+use sp_runtime::DispatchError;
+use sp_std::vec::Vec;
+
+/// Trait for decoding data read from contract memory.
+pub trait Decode {
+	/// The output type to be decoded.
+	type Output: codec::Decode;
+	/// An optional processor, for performing any additional processing on data read from the contract before decoding.
+	type Processor: Processor<Value = Vec<u8>>;
+	/// The error to return if decoding fails.
+	type Error: Get<DispatchError>;
+
+	/// The log target.
+	const LOG_TARGET: &'static str;
+
+	/// Decodes data read from contract memory.
+	///
+	/// # Parameters
+	/// - `env` - The current execution environment.
+	fn decode<E: Ext, S: BufIn>(env: &mut Environment<E, S>) -> Result<Self::Output> {
+		// Charge appropriate weight, based on input length, prior to decoding.
+		// reference: https://github.com/paritytech/polkadot-sdk/blob/117a9433dac88d5ac00c058c9b39c511d47749d2/substrate/frame/contracts/src/wasm/runtime.rs#L267
+		let len = env.in_len();
+		let weight = Schedule::<E::T>::get()
```

Comment:
Deprecated: https://github.com/r0gue-io/pop-node/blob/8072a479a3de951e560a7393e30124f91eede3a4/runtime/devnet/src/extensions.rs#L128C2-L130C60

Reply:
Marked the type alias as deprecated, will be in next push.